### ISSUE TITLE
Rewrite the whole interpreter

### ISSUE THREAD

**Issue by Dandigit** [SimpleBinary/GWN#13]
I've really been meaning to do this since we started the project - so far I've written a simple lexer in Python that I guess we'll build upon with other components.

**Comment by SimpleBinary** [comment]
Yes, I agree, the interpreter is confusing me, and to implement quotation mars is going to be hard.

**Comment by Dandigit** [comment]
I've created a branch over on `Dandigit/GWN` called `complete-rewrite`. Working on pushing the code I've written so far now.

**Comment by SimpleBinary** [comment]
@Dandigit Is it going well? How is the development going? I don't see anything different on Dandigit/GWN  on complete-rewrite

**Comment by SimpleBinary** [comment]
@Dandigit Sorry, I didn't see the new file on complete-rewrite.

**Comment by Dandigit** [comment]
I've just merged #20, which is a temporary sort of half-rewrite. It adds quotation marks, number literals and invalid identifiers. The code is way cleaner, and a new function called `value()` evaluates a given string, returning a string, number, variable or invalid identifier error, preventing tons of code copy-pasting.

**Comment by SimpleBinary** [comment]
### @Dandigit Great, I have hugely updated the datetime function, but I'm wondering what is going to ### be added to the full-rewrite 

##### PS: Can you @mention me in comments so I can find out about them

**Comment by Dandigit** [comment]
The full rewrite is going to mean a lot of things:
- A properly written interpreter, with a real parser
- A codebase that's easy to extend and maintain
- A bigger program, but one that's easier to comprehend
It's going to take some time, hopefully I have a beta up soon, but I think in the long run it'll pay off.
